packages/editor/src/store/actions.js

```diff
@@ -102,43 +195,276 @@ export function editPost( edits ) {
 }
 
 /**
- * Returns an action object to save the post.
- *
- * @param {Object}  options          Options for the save.
- * @param {boolean} options.isAutosave Perform an autosave if true.
+ * Returns action object produced by the updatePost creator augmented by
+ * an optimist option that signals optimistically applying updates.
  *
+ * @param {Object} edits  Updated post fields.
  * @return {Object} Action object.
  */
-export function savePost( options = {} ) {
+export function __experimentalOptimisticUpdatePost( edits ) {
 	return {
-		type: 'REQUEST_POST_UPDATE',
-		options,
+		...updatePost( edits ),
+		optimist: { id: POST_UPDATE_TRANSACTION_ID },
 	};
 }
 
-export function refreshPost() {
-	return {
-		type: 'REFRESH_POST',
+/**
+ * Action generator for saving the current post in the editor.
+ *
+ * @param {Object} options
+ */
+export function* savePost( options = {} ) {
+	const isEditedPostSaveable = yield select(
+		MODULE_KEY,
```

Comment:
I'm curious whether we should bother with this constant, considering there is precedent via hundreds of other components defined in the editor module referencing `'core/editor'` directly.

Reply:
I think having the constant helps prevent against typos personally (however I do understand that reasoning can't be extrapolated to the store/module/reducer key being referenced outside of the package).

Another nit is that I see a variety of terms used to reference this argument in the codebase, sometimes its "reducer key", sometimes its "module key" and sometimes it is "store key".  I just went with `MODULE_KEY` here because that was the variable name given this originally when registering the store.

Reply:
> Another nit is that I see a variety of terms used to reference this argument in the codebase, sometimes its "reducer key", sometimes its "module key" and sometimes it is "store key". I just went with `MODULE_KEY` here because that was the variable name given this originally when registering the store.

Yes, it's been a lingering problem needing to be addressed. There wasn't already an issue, so I created one at #14150.